pkg/infrastructure/azure/azure.go

```diff
@@ -667,26 +667,37 @@ func (p *Provider) PostDestroy(ctx context.Context, in clusterapi.PostDestroyerI
 		return fmt.Errorf("error creating network client factory: %w", err)
 	}
 
-	// XXX: why is in.Metadata.Azure.ResourceGroupName empty?
-	err = deleteSecurityGroupRule(ctx, &securityGroupInput{
-		resourceGroupName:    fmt.Sprintf("%s-rg", in.Metadata.InfraID),
-		securityGroupName:    fmt.Sprintf("%s-nsg", in.Metadata.InfraID),
-		securityRuleName:     "ssh_in",
-		securityRulePort:     "22",
-		networkClientFactory: networkClientFactory,
-	})
+	// See if a load balancer named ${infraID}-internal-outbound-lb exists.
+	// If it does, this is a private cluster. If it does not, this is a
+	// public cluster and we need to delete the SSH forward rule and
+	// security group.
+	_, err = networkClientFactory.NewLoadBalancersClient().Get(
+		ctx,
+		fmt.Sprintf("%s-rg", in.Metadata.InfraID),
+		fmt.Sprintf("%s-internal-outbound-lb", in.Metadata.InfraID),
+		nil,
+	)
 	if err != nil {
```

Comment:
Should this check be more specific like "*NotFound*" before we delete the SG rule?

Reply:
I'd agree that we should, however I couldn't find a specific error that gets returned. 

https://learn.microsoft.com/en-us/rest/api/load-balancer/load-balancers/get?view=rest-load-balancer-2024-01-01&tabs=HTTP

https://learn.microsoft.com/en-us/rest/api/load-balancer/load-balancers/get?view=rest-load-balancer-2024-01-01&tabs=HTTP#clouderror

We only create this rule for public clusters.  We only delete the rule, not the security group. Looks like I commented incorrectly.  The worst case here is an error is returned about the load balancer that isn't "NotFound", and we still delete the rule, which is what we want anyway. So I don't know that being specific here helps anything.

What do you think?

Reply:
Fine with this.